examples/linear_model/plot_ard.py

```diff
@@ -1,119 +1,146 @@
-plt.figure(figsize=(6, 5))
-plt.title("Weights of the model")
-plt.plot(clf.coef_, color="darkblue", linestyle="-", linewidth=2, label="ARD estimate")
-plt.plot(
-    ols.coef_, color="yellowgreen", linestyle=":", linewidth=2, label="OLS estimate"
+# Fit the regressions
+# -------------------
+import pandas as pd
+from sklearn.linear_model import ARDRegression, LinearRegression, BayesianRidge
+
+olr = LinearRegression()
+brr = BayesianRidge(compute_score=True, n_iter=30)
+ard = ARDRegression(compute_score=True, n_iter=30)
+
+olr.fit(X, y)
+brr.fit(X, y)
+ard.fit(X, y)
+
+df = pd.DataFrame(
+    {
+        "LinearRegression": olr.coef_,
+        "BayesianRidge": brr.coef_,
+        "ARDRegression": ard.coef_,
+        "true weights": w,
+    }
 )
-plt.plot(w, color="orange", linestyle="-", linewidth=2, label="Ground truth")
-plt.xlabel("Features")
-plt.ylabel("Values of the weights")
-plt.legend(loc=1)
-
-plt.figure(figsize=(6, 5))
-plt.title("Histogram of the weights")
-plt.hist(clf.coef_, bins=n_features, color="navy", log=True)
-plt.scatter(
-    clf.coef_[relevant_features],
-    np.full(len(relevant_features), 5.0),
-    color="gold",
-    marker="o",
-    label="Relevant features",
+# %%
+# Plot the true and estimated weights
+# -----------------------------------
+import matplotlib.pyplot as plt
+import seaborn as sns
+from matplotlib.colors import LogNorm
+
+plt.figure(figsize=(10, 6))
+ax = sns.heatmap(
+    df.T,
+    norm=LogNorm(),
+    cbar_kws={"label": "weight value"},
+    cmap="rocket_r",
 )
-plt.ylabel("Features")
-plt.xlabel("Values of the weights")
-plt.legend(loc=1)
-
-plt.figure(figsize=(6, 5))
-plt.title("Marginal log-likelihood")
-plt.plot(clf.scores_, color="navy", linewidth=2)
+plt.ylabel("predictor")
+plt.xlabel("coeficients")
+plt.tight_layout(rect=(0, 0, 1, 0.95))
+_ = plt.title("Weights as a function of the model")
+# %%
+# Plot the marginal log-likelihood
+# --------------------------------
+plt.plot(ard.scores_, color="navy", label="Polynomial ARD")
+plt.plot(brr.scores_, color="red", label="Polynomial BayesianRidge")
 plt.ylabel("Score")
 plt.xlabel("Iterations")
+plt.xlim(1, 30)
+plt.legend()
+_ = plt.title("Convergence of marginal log-likelihood")
+# %%
+# Plotting polynomial regressions with std errors of the scores
+# -------------------------------------------------------------
+import numpy as np
+from sklearn.pipeline import make_pipeline
+from sklearn.preprocessing import PolynomialFeatures, StandardScaler
+
+rng = np.random.RandomState(0)
+n_samples = 100
+data_max, data_min = 0, 10
+len_data = data_max - data_min
+# sort the data to make plotting easier later
+data = np.sort(rng.rand(n_samples) * len_data - len_data)
+noise = rng.normal(0, 1, n_samples) * 1.35
+target = np.sqrt(data) * np.sin(data) + noise
+full_data = pd.DataFrame({"input_feature": data, "target": target})
+# X should be 2D for sklearn: (n_samples, n_features)
+data = data.reshape((-1, 1))
```

Comment:
I think this can be simplified.

Can also `data` be renamed to `X` in this example to match conventions?
```suggestion
rng = np.random.RandomState(0)
n_samples = 100

# sort the data to make plotting easier later
# X should be 2D for sklearn: (n_samples, n_features)
data = np.sort(rng.rand(n_samples, 1) * 10)
noise = rng.normal(0, 1, n_samples) * 1.35
target = np.sqrt(data) * np.sin(data) + noise
full_data = pd.DataFrame({"input_feature": data, "target": target})

```